File: server-mh/src/Match/Handler/Models/Board.ts
import { Cell } from "./Cell";
import { BoardConfig } from "./BoardConfig";
import { PlayerColor } from "../Enums";
export class Board {
    public cells: Cell[];

    public config: BoardConfig;

    constructor(config: BoardConfig) {
        this.config = config;
        this.cells = [];
    }// new Board(BoardConfigs.ClassicLudo())

    public CreateBoard(): void {

        this.cells = Array.from(
            { length: this.config.numOfCellsInBoard },
            (_, index) => new Cell(index)
        );

        for (const playerColor in this.config.playerPath) {

            const path = this.config.playerPath[playerColor as unknown as PlayerColor];

            for (const cellIndex of path.initialCells) {
                this.cells[cellIndex].isInitial = true;
            }

            const finalCellIndex = path.homeCells[path.homeCells.length - 1];

            if (finalCellIndex >= 0) {
                this.cells[finalCellIndex].isFinal = true;
            }
        }

        for (const cellIndex of this.config.safeCellsCapability) {
            this.cells[cellIndex].canBecomeSafeCell = true;
        }

        for (const cellIndex of this.config.penaltyCellCapability) {
            this.cells[cellIndex].canBecomePenaltyCell = true;
        }
    }
}

File: Client-unity/Ludo/Assets/Scripts/OfflineMatchHandler/Board.cs
namespace Ludo.Offline
{
    public class Board
    {
        public Cell[] Cells { get; private set; }

        public BoardConfig Config { get; }

        public Board(BoardConfig config)
        {
            Config = config;
            CreateBoard();
        }

        private void CreateBoard()
        {
            Cells = new Cell[Config.NumOfCellsInBoard];

            for (int i = 0; i < Cells.Length; i++)
            {
                Cells[i] = new Cell(i);
            }

            foreach (var playerPath in Config.PlayerPath.Values)
            {
                foreach (var cellIndex in playerPath.InitialCells)
                {
                    Cells[cellIndex].IsInitial = true;
                }

                Cells[playerPath.StartHomeEntryCell].IsSafe = true;

                int finalCellIndex =
                    playerPath.HomeCells[^1];

                if (finalCellIndex >= 0)
                {
                    Cells[finalCellIndex].IsFinal = true;
                }
            }
        }
    }
}